Jenkinsfile

```diff
@@ -144,6 +141,9 @@ def void mvn() {
            -Dcommit.id=$GIT_COMMIT \
            -Dbuild.type=$BUILD_TYPE \
            -Dorg.eclipse.justj.p2.manager.build.url=$JOB_URL \
+           -B sonar:sonar \
```

Comment:
I believe the sonar:sonar invocation has to be done after `verify`. Otherwise there are no binaries available for the tool to analyze.

Reply:
Oh, you mean there needs to be a separate maven invocation?

Reply:
No, I think you just need to move it down:

```
         mvn \
           $promotion_argument \
           --no-transfer-progress \
           -DapiBaselineTargetDirectory=${WORKSPACE} \
           -Dmaven.repo.local=$WORKSPACE/.m2/repository \
           -Dproject.build.sourceEncoding=UTF-8 \
           -Dbuild.id=${BUILD_TIMESTAMP} \
           -Dcommit.id=$GIT_COMMIT \
           -Dbuild.type=$BUILD_TYPE \
           -Dorg.eclipse.justj.p2.manager.build.url=$JOB_URL \
           -B \
           -Dsonar.projectKey=gef-classic \
           -Dsonar.organization=eclipse \
           clean \
           verify \
           sonar:sonar
```

Then the `verify` goal is executed before the `sonar` goal.

Reply:
I see. I thought it was associated with the -B!

Reply:
> I see. I thought it was associated with the -B!

That just enables batch-mode ;D
`-B,--batch-mode	Run in non-interactive (batch) mode (disables output color)`

Reply:
The build is progressing further, so I think this is good now....

Reply:
> That just enables batch-mode ;D
> `-B,--batch-mode Run in non-interactive (batch) mode (disables output color)`

ah now I now what the B stands for. I always had that where there was sonar.

Reply:
You learn something new every day.  😁